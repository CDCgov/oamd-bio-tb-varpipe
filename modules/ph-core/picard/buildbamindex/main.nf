process PICARD_BUILDBAMINDEX {

    tag "${meta.id}"
    label 'process_medium'

    container 'oamd-bio-biocontainers-picard:3.3.0_47ba639_v2'

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*.bai"), emit: bai
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
        picard BuildBamIndex \\
            INPUT=${input} \\
            VALIDATION_STRINGENCY=${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        picard: \$(picard CreateSequenceDictionary --version 2>&1 | grep -o 'Version:.*' | cut -f2- -d:)
    END_VERSIONS
    """
    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.bai

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        picard: \$(picard CreateSequenceDictionary --version 2>&1 | grep -o 'Version:.*' | cut -f2- -d:)
    END_VERSIONS
    """
}
