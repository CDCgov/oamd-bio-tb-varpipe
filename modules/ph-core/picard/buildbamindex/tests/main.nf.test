nextflow_process {

    name "Test Process PICARD_BUILDBAMINDEX"
    script "../main.nf"
    process "PICARD_BUILDBAMINDEX"
    config './nextflow.config'

    tag "modules"
    tag "modules_phcore"
    tag "picard"
    tag "picard/buildbamindex"

    test("tuberculosis fasta") {

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
				    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("tuberculosis fasta -stub") {

        options "-stub"

        when {
            process {
                """
                // Prepare input data
                input[0] = [
                    [ id:'test' ], // meta map
				    file(params.nfcore_modules_testdata_base_path + '/genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
