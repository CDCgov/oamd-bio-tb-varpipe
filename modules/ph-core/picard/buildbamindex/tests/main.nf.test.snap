{
    "tuberculosis fasta -stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f42544a67205d10997289e2413f64ecc"
                ],
                "bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f42544a67205d10997289e2413f64ecc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T11:42:38.821533"
    },
    "tuberculosis fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.paired_end.sorted.bai:md5,704c10dd1326482448ca3073fdebc2f4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f42544a67205d10997289e2413f64ecc"
                ],
                "bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.paired_end.sorted.bai:md5,704c10dd1326482448ca3073fdebc2f4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f42544a67205d10997289e2413f64ecc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T11:51:47.567004"
    }
}
